src/Resources/contao/drivers/DC_Folder.php

```diff
@@ -1126,6 +1133,11 @@ public function move($blnIsAjax=false)
 			// Redirect or reload
 			if (!$objUploader->hasError())
 			{
+				if ($blnIsAjax)
+				{
+					throw new ResponseException(new Response(\Message::generateUnwrapped(), 201));
```

Comment:
I disagree. Please read carefully.

[RFC 2616](https://www.w3.org/Protocols/rfc2616/rfc2616.html) (*Hypertext Transfer Protocol -- HTTP/1.1*)

> The request has been fulfilled and resulted in a new resource being created. The newly created resource can be referenced by the URI(s) returned in the entity of the response, with the most specific URI for the resource given by a Location header field. The response SHOULD include an entity containing a list of resource characteristics and location(s) from which the user or user agent can choose the one most appropriate. The entity format is specified by the media type given in the Content-Type header field. 

RFC 2119 (*Key words for use in RFCs to Indicate Requirement Levels*):
> **SHOULD**   This word, or the adjective "RECOMMENDED", mean that there
   may exist valid reasons in particular circumstances to ignore a
   particular item, but the full implications must be understood and
   carefully weighed before choosing a different course.


1. the valid reason to not include the list is that there can be multiple, so it can't be represented with a `Location` field. Except we could return the `Location` of the parent folder, but I think that's unneccessary.
2. There is actually returned a list of entities. The media type is `text/html`, and the returned list are HTML strings (the messages) of the uploads.